tools/update-deps

```diff
@@ -0,0 +1,129 @@
+
+usage(0) if $help;
+
+my $scriptname = __FILE__;
+my $yamlfile   = "dependencies.yaml";
+my $file       = "$Bin/../$yamlfile";
+my $specfile   = "$Bin/../openQA.spec";
+my $cpanfile   = "$Bin/../cpanfile";
+
+my $data = YAML::PP->new->load_file($file);
+my $spec = path($specfile)->slurp;
+
+my @spectargets = qw(assetpack build client common devel main test worker);
+my @cpantargets = qw(assetpack       client common       main test worker);
+
+update_spec();
+update_cpanfile();
+
+sub update_spec {
+    for my $target (@spectargets) {
+        say "Target $target";
+        my $deps     = $data->{$target . '_requires'};
+        my $prefix   = "%define ${target}_requires";
+        my $specline = $prefix;
+        for my $key (sort keys %$deps) {
+            my $version = $deps->{$key};
+            if (ref $version) {
+                $version = $version->{rpm};
+            }
+            $specline .= " $key";
+            if ($deps->{$key}) {
+                $specline .= " $version";
+            }
+        }
+        my $comment = "# The following line is generated from $yamlfile";
+        if ($spec =~ s/^# .*generated.*\n^$prefix.*/$comment\n$specline/m) {
+            next;
+        }
+        # No comment above the line yet
+        unless ($spec =~ s/^$prefix.*/$comment\n$specline/m) {
+            die "/^$prefix/ not found in $specfile";
+        }
+    }
+
+    path($specfile)->spurt($spec);
+    say "Updated $specfile";
+}
+
+sub update_cpanfile {
+    my $cpan = <<"EOM";
+##################################################
+# WARNING
+# This file is autogenerated by $scriptname
+# from $yamlfile
+##################################################
+
+EOM
+    my %modules;
+    my %test;
+    for my $target (@cpantargets) {
+        my $deps = $data->{$target . '_requires'};
+        for my $key (keys %$deps) {
+            my $module = $key;
+            next unless $module =~ s/^perl\((.*)\)$/$1/;
+            my $version = $deps->{$key};
+            if (ref $version) {
+                $version = $version->{perl};
+            }
+            if ($target eq 'test') {
+                $test{$module} = $version;
+            }
+            else {
+                $modules{$module} = $version;
+            }
+        }
+    }
+    # requires 'Archive::Extract', '> 0.7';
+    for my $module (sort keys %modules) {
+        my $version = $modules{$module};
+        my $line    = "requires '$module'";
+        $line .= qq{, '$version'} if $version;
+        $cpan .= "$line;\n";
+    }
+    $cpan .= <<'EOM';
+
+on 'test' => sub {
+EOM
+    for my $module (sort keys %test) {
+        my $version = $test{$module};
+        my $line    = "requires '$module'";
+        $line .= qq{, '$version'} if $version;
+        $cpan .= "    $line;\n";
+    }
+    $cpan .= <<'EOM';
+};
+EOM
+
+    $cpan .= <<'EOM';
+
+feature 'coverage', 'coverage for CI' => sub {
+    requires 'Devel::Cover';
+    requires 'Devel::Cover::Report::Codecov';
+};
+EOM
```

Comment:
I suggest to put this block into a file so that we do not need to ask contributors to change this script in case we change something about dependencies.

Reply:
ok, I added a `tools/cpanfile.additions`